src/main/java/seedu/address/storage/JsonAdaptedPolicy.java

```diff
@@ -0,0 +1,122 @@
+package seedu.address.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.client.policy.CustomDate;
+import seedu.address.model.client.policy.Frequency;
+import seedu.address.model.client.policy.Policy;
+import seedu.address.model.client.policy.PolicyName;
+import seedu.address.model.client.policy.Premium;
+
+/**
+ * Jackson-friendly version of {@Link Policy}
+ */
+public class JsonAdaptedPolicy {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Policy's %s field is missing!";
+    private final String policyName;
+    private final String startDate;
+    private final String premium;
+    private final String frequency;
+
+    /**
+     * Constructs a {@code JsonAdaptedPolicy} with the given policy details.
+     * @param policyName The name of the policy.
+     * @param startDate The starting date of the policy.
+     * @param premium The value of the policy.
+     * @param frequency The duration of the policy.
+     */
+    @JsonCreator
+    public JsonAdaptedPolicy(@JsonProperty("policyName") String policyName, @JsonProperty("startDate") String startDate,
+                             @JsonProperty("premium") String premium, @JsonProperty("frequency") String frequency) {
+        this.policyName = policyName;
+        this.startDate = startDate;
+        this.premium = premium;
+        this.frequency = frequency;
+    }
+
+    /**
+     * Coverts a given {@code Policy} into this class for Jackson use.
+     * @param source The policy that we are interested in.
+     */
+    public JsonAdaptedPolicy(Policy source) {
+        policyName = source.getPolicyName().toString(); // does this break Law of Demeter?
+        startDate = source.getStartDate().toString();
+        premium = source.getPremium().toString();
+        frequency = source.getFrequency().toString();
+    }
+    /*
+     need tell yuze to make all the new classes to support constructor
+     that takes in strings because i need convert from json to there
+     */
+
+
+    /* Need to see how this works ... based on tracing tag, this seems to get
+    * called only when you update it? that is from object->Json
+    */
+    /*
+    @JsonValue
+    public String getPolicy() {
+        String policy = policyName + ", "
+        StringBuilder policy = new StringBuilder("[");
+        policy.append(policyName);
+        policy.append(", ");
+        policy.append(startDate);
+        policy.append(", ");
+        policy.append(premium);
+        policy.append(", ");
+        policy.append(frequency);
+        /* how to better add tags here ?"
+        policy.append("]");
+        return policy.toString();
```

Comment:
Will this code be used in the future?

Reply:
Might consider in the future, will update!